engine/jobmaster/dm/checkpoint/agent.go

```diff
@@ -61,30 +61,34 @@ var NewCheckpointAgent = NewAgentImpl
 type Agent interface {
 	Create(ctx context.Context) error
 	Remove(ctx context.Context) error
-	Update(ctx context.Context, cfg *config.JobCfg) error
 	IsFresh(ctx context.Context, workerType framework.WorkerType, task *metadata.Task) (bool, error)
 }
 
 // AgentImpl implements Agent
 type AgentImpl struct {
-	mu     sync.RWMutex
-	cfg    *config.JobCfg
-	logger *zap.Logger
+	jobStore *metadata.JobStore
+	logger   *zap.Logger
 }
 
 // NewAgentImpl creates a new AgentImpl instance
-func NewAgentImpl(jobCfg *config.JobCfg, pLogger *zap.Logger) Agent {
+func NewAgentImpl(jobStore *metadata.JobStore, pLogger *zap.Logger) Agent {
 	c := &AgentImpl{
-		logger: pLogger.With(zap.String("component", "checkpoint_agent")),
-		cfg:    jobCfg,
+		logger:   pLogger.With(zap.String("component", "checkpoint_agent")),
+		jobStore: jobStore,
 	}
 	return c
 }
 
-func (c *AgentImpl) getConfig() *config.JobCfg {
-	c.mu.RLock()
-	defer c.mu.RUnlock()
-	return c.cfg
+func (c *AgentImpl) getConfig(ctx context.Context) (*config.JobCfg, error) {
+	state, err := c.jobStore.Get(ctx)
+	if err != nil {
+		return nil, err
+	}
+	job := state.(*metadata.Job)
+	for _, task := range job.Tasks {
+		return (*config.JobCfg)(task.Cfg), nil
+	}
+	return nil, errors.New("no task found in job")
```

Comment:
Currently where's the code that we make sure the job is written to `jobStore` before `getConfig`? 😵

Reply:
We put task config to metadata before create checkpoint in InitImpl now. https://github.com/pingcap/tiflow/blob/7fbc7902689bfebb2cc9d184a4729bbfeaa50615/engine/jobmaster/dm/dm_jobmaster.go#L119

Reply:
Pause job config to checkpoint agent instead of using jobStore in [0f11c31](https://github.com/pingcap/tiflow/pull/6706/commits/0f11c3164e1347985ab35eeef397712c6c612303) now